alacritty/src/input.rs

```diff
@@ -481,13 +481,23 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
         let mut point = self.ctx.size_info().pixels_to_coords(mouse.x, mouse.y);
         point.line = min(point.line, self.ctx.terminal().grid().num_lines() - 1);
 
+        // Reset click state in mouse mode
+        if !self.ctx.modifiers().shift() && self.ctx.mouse_mode() {
+            self.ctx.mouse_mut().click_state = ClickState::None;
+        }
```

Comment:
I don't quite understand this part. Why would we even get to this point, just to reset it?

It seems like what we actually want is to always execute the same code when mouse mode is active, which is the `self.mouse_report(code, ElementState::Pressed);` one?

It seems like restructuring the method could lead to significant improvements in readability here.

Reply:
You can change state during triple/double click, so instead of having it everywhere for mouse mode, I've decided to put it here, doing it in `self.mouse_report(code, ElementState::Pressed);` doesn't work in the mentioned scenario, since your first click could be in normal mode, which won't trigger that reset, but the second/third one in mouse mode.